lib/Transforms/Scalar/CrabLowerIsDeref.cc

```diff
@@ -0,0 +1,115 @@
+#include "seahorn/Analysis/SeaBuiltinsInfo.hh"
+#include "seahorn/Passes.hh"
+#include "seahorn/Support/SeaDebug.h"
+#include "seahorn/Support/SeaLog.hh"
+#include "seahorn/Support/Stats.hh"
+#include "seahorn/Transforms/Instrumentation/CrabAnalysisPass.hh"
+
+#include "llvm/IR/DebugInfo.h"
+#include "llvm/IR/DebugLoc.h"
+
+#include "clam/CfgBuilder.hh"
+#include "clam/Clam.hh"
+
+using namespace llvm;
+using namespace seahorn;
+
+namespace {
+
+struct CrabLowerIsDeref : public FunctionPass {
+private:
+  Value *crabLowerIsDereferenceable(CallBase *IsDerefCall);
+  const llvm::ConstantRange getCrabInstRng(const llvm::Instruction &I);
+
+  std::shared_ptr<clam::InterGlobalClam> m_crab_ptr;
+
+public:
+  static char ID;
+
+  CrabLowerIsDeref() : FunctionPass(ID) {}
+
+  bool runOnFunction(Function &F) override;
+  void getAnalysisUsage(AnalysisUsage &AU) const override {
+    AU.addRequired<CrabAnalysisPass>();
+    AU.addRequired<SeaBuiltinsInfoWrapperPass>();
+    AU.setPreservesAll();
+  }
+  StringRef getPassName() const override { return "CrabLowerIsDeref"; }
+};
+char CrabLowerIsDeref::ID = 0;
+
+} // namespace
+
+bool CrabLowerIsDeref::runOnFunction(Function &F) {
+  LOG("crab-isderef", errs()
+                          << "Start Running Crab lowering is_deref checks\n";);
+  m_crab_ptr = getAnalysis<CrabAnalysisPass>().getCrab();
```

Comment:
Don't use the pass manager to get a Crab analysis. (see my comments in `CrabAnalysisPass.cc`